src/Umbraco.Web.BackOffice/Mapping/ContentMapDefinition.cs

```diff
@@ -199,13 +199,21 @@ private IEnumerable<string> GetActions(IContent source)
         private UrlInfo[] GetUrls(IContent source)
         {
             if (source.ContentType.IsElement)
+            {
                 return Array.Empty<UrlInfo>();
+            }
 
             var umbracoContext = _umbracoContextAccessor.UmbracoContext;
 
-            var urls = umbracoContext == null
-                ? new[] { UrlInfo.Message("Cannot generate URLs without a current Umbraco Context") }
-                : source.GetContentUrls(_publishedRouter, umbracoContext, _localizationService, _localizedTextService, _contentService, _variationContextAccessor, _loggerFactory.CreateLogger<IContent>(), _uriUtility, _publishedUrlProvider).ToArray();
+            if (umbracoContext == null)
+            {
+                return new[] { UrlInfo.Message("Cannot generate URLs without a current Umbraco Context") };
+            }
+
+            // NOTE: unfortunately we're not async, we'll use .Result and hope this won't cause a deadlock anywhere for now
+            var urls = source.GetContentUrlsAsync(_publishedRouter, umbracoContext, _localizationService, _localizedTextService, _contentService, _variationContextAccessor, _loggerFactory.CreateLogger<IContent>(), _uriUtility, _publishedUrlProvider)
+                .Result
```

Comment:
Is it safe ? Should we use GetAwaiter().GetResult() instead?

Reply:
In some cases it won't matter and it's never 100% safe. It will be safe on request threads but it won't be safe on threadpool (if maxed out)/ui threads. In some cases you can wrap in a `Task.Run` to avoid a deadlock but that all depends on where the code is called from or jump through other hoops. For now we'll do `.ConfigureAwait(false).GetAwaiter().GetResult()` so at least any exception is unwrapped and we're not capture the sync context. 

The true solution is to either make the mapper async (probably best) or create native sync methods like GetContentUrls but to do that will also require making native sync methods of the new event aggregator.